plugins/org.jboss.tools.quarkus.ui/src/org/jboss/tools/quarkus/ui/wizard/CodeProjectTypeWizardPage.java

```diff
@@ -148,10 +157,35 @@ public String getText(Object element) {
             .to(BeanProperties.value(USE_DEFAULT_LOCATION_PROPERTY).observe(model))
             .converting(new InvertingBooleanConverter()).in(dbc);
     ControlDecorationSupport.create(locationBinding, SWT.LEFT | SWT.TOP, null, new RequiredControlDecorationUpdater());
+    
+    dbc.addValidationStatusProvider(new MultiValidator() {
+		
+		@Override
+		protected IStatus validate() {
+		    Tool tool = (Tool) projectTypeObservable.getValue();
+		    String projectName = projectNameObservable.getValue();
+		    String location = txtLocationObservable.getValue();
+		    File   locationFile = new File(location);
+		    
+		    if (Tool.GRADLE.equals(tool)) {
+		    	if (isDirectChildOfWorkspaceRootFolder(location) && !projectName.equals(locationFile.getName())) {
+			    	return ValidationStatus.error("Gradle projects at the root the workspace must be using a folder with the same name");
```

Comment:
please review the error message, seems missing some words?
```suggestion
			    	return ValidationStatus.error("Gradle projects at the root of the workspace must be using a folder with the same name");
```